hypervisor/include/arch/x86/timer.h

```diff
@@ -48,19 +74,63 @@ static inline void initialize_timer(struct hv_timer *timer,
 	}
 }
 
+/**
+ * @brief Check a timer whether expired.
+ *
+ * @param[in] timer Pointer to timer.
+ *
+ * @retval true if the timer is expired, false otherwise.
+ */
 static inline bool timer_expired(const struct hv_timer *timer)
 {
 	return ((timer->fire_tsc == 0UL) || (rdtsc() >= timer->fire_tsc));
 }
 
-/*
- * Don't call add_timer/del_timer in the timer callback function.
+/**
+ * @brief Add a timer.
+ *
+ * @param[in] timer Pointer to timer.
+ *
+ * @retval
```

Comment:
While you could do that (using multiple lines and bullet points) with `return`, it does not render correctly when using `retval`. Can you change this to:
```
@retval 0 on success
@retval -EINVAL \p timer has an invalid value
```

Reply:
Done